pkg/arvo/sys/hoon.hoon

```diff
@@ -1641,25 +1637,23 @@
         b
       ?:  (mor p.n.a p.n.b)
         ?:  =(p.n.b p.n.a)
-          [n.b $(a l.a, b l.b) $(a r.a, b r.b)]
+          :+  [p.n.a (meg p.n.a q.n.a q.n.b)]
```

Comment:
I don't like that the `=(p.n.b p.n.a)` check is only run if `+mor` produces `%.y`.  Seems somewhat brittle w.r.t. the details of `+mor`.  If it started to behave more like `+lth` than `+lte`, could that change the behavior here?  If the keys are equal, I don't think we should care about where exactly `+mor` switches from `%.n` to `%.y`.

I think we should first check equality, and then only run `+mor` if the keys aren't equal.

Reply:
That makes sense. Philip expressed similar concerns when we talked about this issue. I'll update the `+uni` arms as well

Reply:
I have just pushed this change. I believe I have addressed your other comments as well, but let me know if i missed something. Thanks!